### ISSUE TITLE
special char results in Index and count must refer to a location within the string

### ISSUE THREAD

**Issue by OneCyrus** [neosmart/SecureStore#11]
using special chars in the string result in the following error. e.g. `@lt€r` when setting through the client

```
Test method Tests.FunctionalityTests.StoreAndLoad threw exception: 
System.ArgumentOutOfRangeException: Index and count must refer to a location within the string. (Parameter 's')
```

**Comment by OneCyrus** [comment]
I added a PR with the reproduction here: https://github.com/neosmart/SecureStore/pull/12

not sure what is the best way to resolve this.

**Comment by mqudsi** [comment]
Do you have the full stack trace? I wonder if it’s not so much a special character so much as an invalid one..

**Comment by mqudsi** [comment]
Fixed in 249a532b41e97dc506a73665b3f2c5e50a6e6960.

Thanks for finding this. I was passing the byte length to a parameter that expected the character count.
I'll push a release to nuget soon, but you can build from master in the meantime.

**Comment by OneCyrus** [comment]
perfect. thank you!
